server/routes/star.js

```diff
@@ -24,54 +24,61 @@ router.get("/:starId", async (req, res) => {
     }
 
     try {
-        const id = Number(starId)
+        const id = Number(starId);
         const stars = await pool.query("SELECT * FROM stars WHERE id = $1", [id]);
-        if (stars.rows.length === 0){
-            return res.status(404).json({msg: "the star not found", status:"error" })
+        if (stars.rows.length === 0) {
+            return res.status(404).json({ msg: "the star not found", status: "error" })
         }
-        res.status(200).json(stars.rows[0])
+        res.status(200).json(stars.rows[0]);
     } catch (error) {
-        res.status(500).json(error)
+        res.status(500).json(error);
     }
 })
 
 router.post("/", async (req, res) => {
     const { title, competence, situation, task, action, result } = req.body;
-
     // validations
     if (!title || !situation || !task || !action || !result) {
         return res.status(400).json({ msg: "please provide mandatory fields", status: "error" })
     }
 
-
     try {
-        const currentDate = new Date()
-        const values = [title, competence, situation, task, action, result, currentDate, currentDate]
-        const star = await pool.query("INSERT INTO stars (title, competence, situation, task, action, result, created_at, updated_at) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *", values)
-        res.status(200).json(star.rows[0])
+        const currentDate = new Date();
+        const values = [title, competence, situation, task, action, result, currentDate, currentDate];
+        const userId = req.session.passport?.user;
+        values.push(userId);
+
+        const star = await pool.query("INSERT INTO stars (title, competence, situation, task, action, result, created_at, updated_at, user_id) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9) RETURNING *", values);
```

Comment:
Not an issue but a good thing to know for the future. Usually when setting date fields such as `CREATED_AT` or `UPDATED_AT` we do not want to set them at the application layer but at the data layer. What I mean by this is the for `CREATED_AT` we probably want to have the column default to `TIMESTAMP` and for UPDATED_AT we want to have a trigger on the DB so that everytime the row changed then UPDATED_AT is set to current `TIMESTAMP` https://www.the-art-of-web.com/sql/trigger-update-timestamp/

Reply:
I really appriciate your explanation. I will definitly study about it.

Reply:
No problem, one final thing on dates is that I can see you are using `new Date()`. Throughout your DEV career you will work on apps that exist in many different timezones so as a general rule we should <em>always</em> use UTC timezone. This will save you lots of headaches in future debugging (trust me I know from experience on that one!!)

Reply:
aah , I didnt know that. that was a really good point. thanks.